### PR TITLE
Adds documentation for testing and mocking facts

### PR DESCRIPTION
**Pull request by logicminds**
Every time I create a fact I always forget how exactly to mock the data.  As a result I have added the helpful information here so that myself and others can easily see and use. 

This will be part of the unit test file anytime `pdk new fact <fact_name>` is run.

### CODE REVIEW

File: object_templates/fact_spec.erb [hunk 1]
```diff
@@ -7,9 +7,42 @@ require 'facter/<%= name %>'
...
   subject(:fact) { Facter.fact(<%= name.to_sym.inspect %>) }
 
+  # Mocking Facts
+  # You will most likely need to mock other facts if your custom fact relies up on them
+  # You can mock a existing core fact via:
+  #  allow(Facter.fact(:fqdn)).to receive(:value).and_return('test.example.com')
+  #  allow(Facter).to receive(:value).with(:fqdn).and_return('test.example.com')
+  # This all depends on how you utilize it in the fact code
+
+  #
```
Comment (chelnak) [inline object_templates/fact_spec.erb:17]: Shall we remove this line along with the comment?

File: object_templates/fact_spec.erb [hunk 2]
```diff
@@ -7,9 +7,42 @@ require 'facter/<%= name %>'
...
+  # This mock will go inside your test block or the before each block
+
+  # Mocking confine example
+  # confine kernel: 'Linux' (Located in Fact)
+  # allow(Facter.fact(:kernel)).to receive(:value).and_return('Linux')
+
+  # Using and Mocking Exec
+  # You should always use the Facter Exectution method to execute commands on a system
+  # This will allow you to easily mock items as a result
+  #
```
Comment (chelnak) [inline object_templates/fact_spec.erb:36]: Same goes here. Maybe just have a new line and remove the comment.

**Review by chelnak — DISMISSED** [review]
I love this. Thank you!

Have approved but also left two minor comments.

File: object_templates/fact_spec.erb (hunk 1 above)
Comment (logicminds) [reply object_templates/fact_spec.erb]: updated

File: object_templates/fact_spec.erb (hunk 2 above)
Comment (logicminds) [reply object_templates/fact_spec.erb]: updated

**Review by chelnak — APPROVED** [review]

**Pull request merged**